src/_cffi_src/build_openssl.py

```diff
@@ -38,8 +38,15 @@ def _osx_libraries(build_static):
 
 
 _PRE_INCLUDE = """
+#include <openssl/opensslv.h>
+/*
+    LibreSSL removed e_os2.h from the public headers so we'll only include it
+    if we're using vanilla OpenSSL.
+*/
```

Comment:
Ok, than I think I'd be in favor of simplifying this.

On Mon, Mar 28, 2016 at 10:12 PM, Paul Kehrer <notifications@github.com>
wrote:

> In src/_cffi_src/build_openssl.py
> <https://github.com/pyca/cryptography/pull/2861#discussion_r57663912>:
>
> > @@ -38,8 +38,15 @@ def _osx_libraries(build_static):
> >
> >
> >  _PRE_INCLUDE = """
> > +#include <openssl/opensslv.h>
> > +/*
> > +    LibreSSL removed e_os2.h from the public headers so we'll only include it
> > +    if we're using vanilla OpenSSL.
> > +*/
>
> Nope, _WIN32 is guaranteed defined on 64-bit platforms as well.
> https://msdn.microsoft.com/en-us/library/b0084kay.aspx
>
> —
> You are receiving this because you commented.
> Reply to this email directly or view it on GitHub
> <https://github.com/pyca/cryptography/pull/2861/files/1e88d741f6e15d9170466a08dfe85f28b12c7623#r57663912>
>



-- 
"I disapprove of what you say, but I will defend to the death your right to
say it." -- Evelyn Beatrice Hall (summarizing Voltaire)
"The people's good is the highest law." -- Cicero
GPG Key fingerprint: D1B3 ADC0 E023 8CA6